docs/docs/guides/web3_config/index.md

```diff
@@ -201,6 +202,610 @@ console.log(web3.getContextObject().config)
+```
+:::
+
+#### All available choices for defaultBlock:
+```ts
+web3.defaultBlock = 20167235; // A block number
+web3.defaultBlock = "earliest"; // The genesis block
+web3.defaultBlock = "latest"; // The latest block (current head of the blockchain)
+web3.defaultBlock = "pending"; // The currently mined block (including pending transactions)
+web3.defaultBlock = "finalized"; // (For POS networks) The finalized block is one which has been accepted as canonical by greater than 2/3 of validators
+web3.defaultBlock = "safe"; // (For POS networks) The safe head block is one which under normal network conditions, is expected to be included in the canonical chain. Under normal network conditions the safe head and the actual tip of the chain will be equivalent (with safe head trailing only by a few seconds). Safe heads will be less likely to be reorged than the proof of work network`s latest blocks.
+```
+
+### transactionBlockTimeout
+The `transactionBlockTimeout` is used over socket-based connections. This option defines the amount of new blocks it should wait until the **first confirmation** happens, otherwise the PromiEvent rejects with a timeout error.
+
+The default value of `transactionBlockTimeout` is 50. 
+
+```ts
+import { Web3 } from 'web3';
+
+const web3 = new Web3('http://127.0.0.1:7545');
+
+web3.transactionBlockTimeout = 50;
+
+console.log(web3.getContextObject().config);
+```
+:::info
+The `transactionBlockTimeout` can be configured both globally and at the package level:
+```ts
+import { Web3 } from 'web3';
+
+const web3 = new Web3('http://127.0.0.1:7545');
+
+web3.eth.transactionBlockTimeout = 50;
+
+console.log(web3.eth.getContextObject().config);
+```
+:::
+
+### transactionConfirmationBlocks
+This defines the number of blocks it requires until a transaction is considered confirmed. Different chains have varying security considerations and requirements for confirmation block numbers.
+
+The default value of `transactionConfirmationBlocks` is 24.
+
+```ts
+import { Web3 } from 'web3';
+
+const web3 = new Web3('http://127.0.0.1:7545');
+
+web3.transactionConfirmationBlocks = 24;
+
+console.log(web3.getContextObject().config);
+```
+:::info
+The `transactionConfirmationBlocks` can be configured both globally and at the package level:
+```ts
+import { Web3 } from 'web3';
+
+const web3 = new Web3('http://127.0.0.1:7545');
+
+web3.eth.transactionConfirmationBlocks = 24;
+
+console.log(web3.eth.getContextObject().config);
+```
+:::
+
+
+### transactionPollingInterval
+The `transactionPollingInterval` is used over HTTP connections. This option defines the number of seconds between Web3 calls for a receipt which confirms that a transaction was mined by the network. Modifying this value can reduce the wait time for confirmations or decrease the number of network requests. Setting the `transactionPollingInterval` would also set `transactionReceiptPollingInterval` and `transactionConfirmationPollingInterval` to the same value. 
```

Comment:
Make sure that these links work, but it would be nice to link to the relevant sections here.

```suggestion
The `transactionPollingInterval` is used over HTTP connections. This option defines the number of seconds between Web3 calls for a receipt which confirms that a transaction was mined by the network. Modifying this value can reduce the wait time for confirmations or decrease the number of network requests. Setting the `transactionPollingInterval` would also set [`transactionReceiptPollingInterval`](#transactionReceiptPollingInterval) and [`transactionConfirmationPollingInterval`](#transactionConfirmationPollingInterval) to the same value. 
```